examples/vision/aodnet.py

```diff
@@ -0,0 +1,362 @@
+            kernel_size=7,
+            padding="same",
+            activation=tf.nn.relu,
+            kernel_initializer=tf.keras.initializers.RandomNormal(stddev=stddev),
+            kernel_regularizer=tf.keras.regularizers.L2(weight_decay),
+        )
+        self.conv_layer_5 = tf.keras.layers.Conv2D(
+            filters=3,
+            kernel_size=3,
+            padding="same",
+            activation=tf.nn.relu,
+            kernel_initializer=tf.keras.initializers.RandomNormal(stddev=stddev),
+            kernel_regularizer=tf.keras.regularizers.L2(weight_decay),
+        )
+        self.relu = tf.keras.layers.ReLU(max_value=1.0)
+
+    def call(self, inputs, *args, **kwargs):
+        conv_1 = self.conv_layer_1(inputs)
+        conv_2 = self.conv_layer_2(conv_1)
+        concat_1 = tf.concat([conv_1, conv_2], axis=-1)
+        conv_3 = self.conv_layer_3(concat_1)
+        concat_2 = tf.concat([conv_2, conv_3], axis=-1)
+        conv_4 = self.conv_layer_4(concat_2)
+        concat_3 = tf.concat([conv_1, conv_2, conv_3, conv_4], axis=-1)
+        k = self.conv_layer_5(concat_3)
+        j = k * inputs - k + 1.0
+        output = self.relu(j)
+        return output
+
+
+"""
+In this example, we are training the model using a single GPU (an NVIDIA Tesla
+P100) and the`tf.distribute.OneDeviceStrategy`. This
+[TensorFlow distribution strategy](https://www.tensorflow.org/tutorials/distribute/keras)
+will place any variables created in its scope on the specified device. We will
+define and compile our model in the scope of our distribution strategy, using
+Adam as the optimizer, the Mean Squared Error
+as the loss function, and the PSNR (Peak Signal Noise Ratio) as our metric.
+**Note:** For training the AOD-Net in a multi-GPU environment, we can also use
+`tf.distribute.MirroredStrategy` (go to the
+[Distributed training](https://www.tensorflow.org/guide/distributed_training)
+guide more information).
+"""
+
+
+def peak_signal_noise_ratio(y_true, y_pred):
+    return tf.image.psnr(y_pred, y_true, max_val=255.0)
+
+strategy = tf.distribute.OneDeviceStrategy(device="/gpu:0")
+
+with strategy.scope():
+    model = AODNet(name='AODNet', stddev=0.02, weight_decay=1e-4)
+    model.compile(
+        optimizer=tf.keras.optimizers.Adam(learning_rate=1e-4),
+        loss=tf.keras.losses.MeanSquaredError(),
+        metrics=[peak_signal_noise_ratio]
+    )
+
+"""
+# Training
+
+We train the AOD-Net for 10 epochs. On an NVIDIA Tesla P100, each epoch
+can take around 190 seconds, so the model may take around 30 minutes to train.
+"""
+
+training_history = model.fit(train_dataset, validation_data=val_dataset, epochs=10)
+
+"""
+## Plot of Training and Validation Loss
+"""
+
+plt.plot(training_history.history["loss"])
+plt.plot(training_history.history["val_loss"])
+plt.title("Loss Curve")
+plt.ylabel("loss")
+plt.xlabel("epoch")
+plt.legend(["train", "val"], loc="upper left")
+plt.show()
+
+"""
+## Plot of Training and Validation PSNR
```

Comment:
Suggestion: use sentence case, "PSNRs" (since it is "ratios", it should be plural)

```suggestion
## Plot of training and validation PSNRs
```